services/management/ia/alerts_service_test.go

```diff
@@ -206,3 +220,122 @@ func TestSatisfiesFilters(t *testing.T) {
 		}
 	})
 }
+
+func TestListAlerts(t *testing.T) {
+	ctx := context.Background()
+	sqlDB := testdb.Open(t, models.SkipFixtures, nil)
+	db := reform.NewDB(sqlDB, postgresql.Dialect, reform.NewPrintfLogger(t.Logf))
+
+	t.Cleanup(func() {
+		require.NoError(t, sqlDB.Close())
+	})
+
+	q := db.Querier
+	now := strfmt.DateTime(time.Now())
+
+	tmpl, err := models.CreateTemplate(q, &models.CreateTemplateParams{
+		Template: &alert.Template{
+			Name:     "test_Template",
+			Version:  1,
+			Summary:  gofakeit.Quote(),
+			Expr:     gofakeit.Quote(),
+			Severity: common.Warning,
+		},
+		Source: "USER_FILE",
+	})
+	require.NoError(t, err)
+
+	rule, err := models.CreateRule(q, &models.CreateRuleParams{
+		TemplateName: tmpl.Name,
+		Severity:     models.Severity(common.Warning),
+	})
+	require.NoError(t, err)
+
+	const alertsCount = 25
+	mockAlert := &mockAlertManager{}
+	var mockedAlerts []*ammodels.GettableAlert
+	for i := 0; i < alertsCount; i++ {
+		mockedAlerts = append(mockedAlerts, &ammodels.GettableAlert{
+			Alert: ammodels.Alert{
+				Labels: map[string]string{
+					"ia":        "1",
+					"alertname": rule.ID,
+				},
+			},
+			Fingerprint: pointer.ToString(strconv.Itoa(i)),
+			Status: &ammodels.AlertStatus{
+				State: pointer.ToString("active"),
+			},
+			StartsAt:  &now,
+			UpdatedAt: &now,
+		})
+	}
+	mockAlert.On("GetAlerts", ctx).Return(mockedAlerts, nil)
+
+	tmplSvc := NewTemplatesService(db)
+	tmplSvc.Collect(ctx)
+	svc := NewAlertsService(db, mockAlert, tmplSvc)
+
+	findAlerts := func(alerts []*iav1beta1.Alert, alertIDs ...string) bool {
+		if len(alerts) != len(alertIDs) {
+			return false
+		}
+		m := make(map[string]bool)
+		for _, alertID := range alertIDs {
+			m[alertID] = false
+		}
+		for _, a := range alerts {
+			val, ok := m[a.AlertId]
+			// Extra alert
+			if !ok {
+				return false
+			}
+			// Duplicate
+			if val {
+				return false
+			}
+			m[a.AlertId] = true
+		}
+		for _, v := range m {
+			// AlertID was not in alerts
+			if !v {
+				return false
+			}
+		}
+		return true
+	}
+
+	t.Run("without pagination", func(t *testing.T) {
+		res, err := svc.ListAlerts(ctx, &iav1beta1.ListAlertsRequest{})
+		assert.NoError(t, err)
+		assert.Len(t, res.Alerts, len(mockedAlerts))
```

Comment:
I suppose we can use `ElementsMatch` here

Reply:
`assert.ElementsMatch(t, res.Alerts, mockedAlerts)` fails.

Reply:
Ah, `res.Alerts` and `mockedAlerts` are different types I suppose. But anyway better to check not just elements number, but that it's elements that we expect